docker_ci_deploy/tests/test_main.py

```diff
@@ -124,40 +124,95 @@ def test_image_unparsable(self):
             RegistryTagger('registry:5000').generate_tag(image)
 
 
+class TestGenerateSemverVersionsFunc(object):
+    def test_standard_version(self):
+        """
+        When a standard 3-part semantic version is passed, 3 version strings
+        should be returned with decreasing levels of precision.
+        """
+        versions = generate_semver_versions('5.4.1')
+        assert_that(versions, Equals(['5.4.1', '5.4', '5']))
+
+    def test_extended_version(self):
+        """
+        When a version is passed with extra information separated by '-',
+        version strings should be returned with decreasing levels of precision.
+        """
+        versions = generate_semver_versions('5.5.0-alpha')
+        assert_that(versions, Equals(['5.5.0-alpha', '5.5.0', '5.5', '5']))
```

Comment:
It is but it's kinda debatable. When PyPy3 was on version "5.5.0-alpha" the official Docker image was tagged in this way. I don't think we use this specific behaviour right now, though.

I think it is important to support version strings that aren't just made up of numbers. For example I might want to tag `django-bootstrap` with [`py3.5`, `py3`] and `py3.6`.